From f6c510499cd3845a9d3c276ec0fb034cd52b9a33 Mon Sep 17 00:00:00 2001
From: terrycrhuang <terrycrhuang@google.com>
Date: Fri, 4 May 2018 13:47:06 +0800
Subject: [PATCH 9/9] libril:  Change rild initial sequence to  guarantee
 non-null function pointer before rild register its hidl service

Test: Build and local test
Bug: 78886552
Change-Id: I1137cc0f4f09f0304662a2227b84c2b5b228d45d
---
 ril/libril/ril_service.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ril/libril/ril_service.cpp b/ril/libril/ril_service.cpp
index ab5d8d4..392fae6 100644
--- a/ril/libril/ril_service.cpp
+++ b/ril/libril/ril_service.cpp
@@ -8475,6 +8475,9 @@ void radio::registerService(RIL_RadioFunctions *callbacks, CommandInfo *commands
     simCount = SIM_COUNT;
     #endif
 
+    s_vendorFunctions = callbacks;
+    s_commands = commands;
+
     configureRpcThreadpool(1, true /* callerWillJoin */);
     for (int i = 0; i < simCount; i++) {
         pthread_rwlock_t *radioServiceRwlockPtr = getRadioServiceRwlock(i);
@@ -8495,9 +8498,6 @@ void radio::registerService(RIL_RadioFunctions *callbacks, CommandInfo *commands
         ret = pthread_rwlock_unlock(radioServiceRwlockPtr);
         assert(ret == 0);
     }
-
-    s_vendorFunctions = callbacks;
-    s_commands = commands;
 }
 
 void rilc_thread_pool() {
-- 
2.17.1

